root/client/src/components/table/table.js

```diff
@@ -1,29 +1,179 @@
+          'Saturday': new Date().setDate(new Date().getDate() + 3),
+        }
+        break;
+      case 4:
+        newStateWeek = {
+          'Sunday': new Date().setDate(new Date().getDate() - 4),
+          'Monday': new Date().setDate(new Date().getDate() - 3),
+          'Tuesday': new Date().setDate(new Date().getDate() - 2),
+          'Wednesday': new Date().setDate(new Date().getDate() - 1),
+          'Thursday': today[Symbol.toPrimitive]('number'),
+          'Friday': new Date().setDate(new Date().getDate() + 1),
+          'Saturday': new Date().setDate(new Date().getDate() + 2),
+        }
+        break;
+      case 5:
+        newStateWeek = {
+          'Sunday': new Date().setDate(new Date().getDate() - 5),
+          'Monday': new Date().setDate(new Date().getDate() - 4),
+          'Tuesday': new Date().setDate(new Date().getDate() - 3),
+          'Wednesday': new Date().setDate(new Date().getDate() - 2),
+          'Thursday': new Date().setDate(new Date().getDate() -1),
+          'Friday': today[Symbol.toPrimitive]('number'),
+          'Saturday': new Date().setDate(new Date().getDate() + 1),
+        }
+        break;
+      case 6:
+        newStateWeek = {
+          'Sunday': new Date().setDate(new Date().getDate() - 6),
+          'Monday': new Date().setDate(new Date().getDate() - 5),
+          'Tuesday': new Date().setDate(new Date().getDate() - 4),
+          'Wednesday': new Date().setDate(new Date().getDate() - 3),
+          'Thursday': new Date().setDate(new Date().getDate() - 2),
+          'Friday': new Date().setDate(new Date().getDate() - 1),
+          'Saturday': today[Symbol.toPrimitive]('number'),
+        }
+        break;
+      // Need to change default
+      default: {
+        newStateWeek = {
+          'Sunday': new Date().setDate(new Date().getDate() - 1),
+          'Monday': today[Symbol.toPrimitive]('number'),
+          'Tuesday': new Date().setDate(new Date().getDate() + 1),
+          'Wednesday': new Date().setDate(new Date().getDate() + 2),
+          'Thursday': new Date().setDate(new Date().getDate() + 3),
+          'Friday': new Date().setDate(new Date().getDate() + 4),
+          'Saturday': new Date().setDate(new Date().getDate() + 5),
+        }
+        break;
+      }
+    }
+
+    setWeek(newStateWeek);
+  }, []);
+
+  // 7 * 24 * 60 * 60 * 1000
+
+  const handlePrevWeek = (e) => {
+    e.preventDefault();
+    // Take each day of this.state.week. Move that back 7 days.
+    const newStateWeek = {
+      'Sunday': week['Sunday'] - (7 * 24 * 60 * 60 * 1000),
+      'Monday': week['Monday'] - (7 * 24 * 60 * 60 * 1000),
+      'Tuesday': week['Tuesday'] - (7 * 24 * 60 * 60 * 1000),
+      'Wednesday': week['Wednesday'] - (7 * 24 * 60 * 60 * 1000),
+      'Thursday': week['Thursday'] - (7 * 24 * 60 * 60 * 1000),
+      'Friday': week['Friday'] - (7 * 24 * 60 * 60 * 1000),
+      'Saturday': week['Saturday'] - (7 * 24 * 60 * 60 * 1000),
+    }
```

Comment:
similar to the above, we can make this better with an algo. here's a potential optimization:


```suggestion
    const newStateWeek = {};
    for(let day in week) {
      newStateWeek[day] = week[day] - (7 * 24 * 60 * 60 * 1000);
    }
```

^^ the above is untested, but you get the idea

Reply:
Made this change.